pkg/controller/bastion/actuator_reconcile.go

```diff
@@ -121,6 +131,56 @@ func (a *actuator) Reconcile(ctx context.Context, bastion *extensionsv1alpha1.Ba
 	return a.Client().Status().Patch(ctx, bastion, patch)
 }
 
+func getNetAndSubnet(ctx context.Context, a *actuator, cluster *extensions.Cluster) (vNet, subnet string, err error) {
+	var infrastructureStatus *gcp.InfrastructureStatus
+	worker := &extensionsv1alpha1.Worker{}
+
+	err = a.Client().Get(ctx, client.ObjectKey{Namespace: cluster.ObjectMeta.Name, Name: cluster.Shoot.Name}, worker)
+	if err != nil {
+		return "", "", err
+	}
+
+	if worker == nil || worker.Spec.InfrastructureProviderStatus == nil {
+		return "", "", errors.New("infrastructure provider status must be not empty for worker")
+	}
+
+	if infrastructureStatus, err = helper.InfrastructureStatusFromRaw(worker.Spec.InfrastructureProviderStatus); err != nil {
+		return "", "", err
+	}
+
+	if infrastructureStatus.Networks.VPC.Name == "" {
+		return "", "", errors.New("virtual network must be not empty for infrastructure provider status")
+	}
+
+	if len(infrastructureStatus.Networks.Subnets) == 0 {
+		return "", "", errors.New("subnet must be not empty")
+	}
+
+	var purposeNodesSet = sets.NewString()
+	for _, subnet := range infrastructureStatus.Networks.Subnets {
+		purposeNodesSet.Insert(string(subnet.Purpose))
+	}
+
+	if !purposeNodesSet.Has(string(gcp.PurposeNodes)) {
+		return "", "", errors.New("none of subenet purpose as node found")
```

Comment:
Otherwise small nit: `subenet` typo here

Reply:
commit in [1766ec8](https://github.com/gardener/gardener-extension-provider-gcp/pull/436/commits/1766ec88d56f71912d6a0aee977245f7ad01ba46)